src/mlpack/core/optimizers/sgd/sgd.hpp

```diff
@@ -99,6 +102,26 @@ class SGD
       const bool shuffle = true);
 
   /**
+   * Overloaded SGD optimizer constructor with UpdateType parameter.
+   *
+   * @param function
+   * @param updatePolicyType The  update policy used for the gradient update
+   *     during the iterations.
+   * @param stepSize Step size for each iteration.
+   * @param maxIterations Maximum number of iterations allowed (0 means no
+   *     limit).
+   * @param tolerance Maximum absolute tolerance to terminate algorithm.
+   * @param shuffle If true, the function order is shuffled; otherwise, each
+   *     function is visited in linear order.
+   */
+  SGD(DecomposableFunctionType& function,
+      UpdatePolicyType updatePolicyType,
```

Comment:
I would actually suggest making this the last parameter and giving it a default: `UpdatePolicyType updatePolicyType = UpdatePolicyType()`.  That would save you from needing to have two different constructor overloads.